instance/tasks.py

```diff
@@ -339,3 +344,111 @@ def cleanup_old_betatest_users():
         users_to_delete = queryset.filter(is_active=False, profile__modified__lte=delete_cutoff)
         logger.info('Deleting %s users.', users_to_delete.count())
         users_to_delete.delete()
+
+
+class KillZombiesRunner:
+    """
+    Helper class to run `kill_zombies_periodically`
+    """
+
+    def __init__(self):
+        self.region: str = getattr(settings, "OPENSTACK_REGION", "gra")
+        self.recipient: list = [email for _, email in getattr(settings, "ADMINS", [])]
+        self.threshold: int = getattr(settings, "KILL_ZOMBIES_WARNING_THRESHOLD")
+
+    def send_email(self, subject: str, body: str) -> int:
+        """
+        Utility method for sending emails
+        """
+        if self.recipient:
+            email = EmailMessage(
+                subject, body, settings.DEFAULT_FROM_EMAIL, self.recipient
+            )
+            return email.send()
+        else:
+            logging.info("Email recipient is undefined. Skipping email.")
+            return None
+
+    def trigger_warning(self):
+        """
+        Warns when more than `threshold` VMs will be terminated
+        """
+        subject = "{} Zombie instances over current threshold ({})".format(
+            datetime.now().strftime("%Y-%m-d"), self.threshold
+        )
+        body = (
+            "The number of instances in region {} to be terminated"
+            "is over KILL_ZOMBIES_WARNING_THRESHOLD ({}).\n"
```

Comment:
```suggestion
            "The number of zombie OpenStack VMs to be terminated in the {} region "
            "is over the KILL_ZOMBIES_WARNING_THRESHOLD of {}.\n"
```

```suggestion
            "The number of zombie OpenStack VMs to be terminated in the {} region  "
```
Note the trailing space which is needed during string concatenation to separate the words.

Reply:
Applied. Thanks for the whitespace addendum.